### PR TITLE
chore: update api_doc.mustache and model_doc.mustache

### PR DESCRIPTION
**Pull request by thinkingserious**
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license

Internal reference: DI-1233
Related PR: https://github.com/twilio/twilio-go/pull/67

### CODE REVIEW

File: src/main/resources/twilio-go/api_doc.mustache [hunk 1]
```diff
@@ -14,11 +14,45 @@ Method | HTTP request | Description
 
 ## {{{operationId}}}
 
-> {{#returnType}}{{{returnType}}} {{/returnType}}{{{operationId}}}(ctx, {{#allParams}}{{#required}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{/required}}{{/allParams}}{{#hasOptionalParams}}optional{{/hasOptionalParams}})
+> {{#returnType}}{{{.}}} {{/returnType}}{{{operationId}}}(ctx{{#pathParams}}, {{paramName}}{{/pathParams}}){{#allParams}}{{^isPathParam}}.{{vendorExtensions.x-export-param-name}}({{paramName}}){{/isPathParam}}{{/allParams}}.Execute()
```
Comment (shwetha-manvinkurke) [inline src/main/resources/twilio-go/api_doc.mustache:17]: This is incorrect. The generated [method](https://github.com/twilio/twilio-go/blob/main/twilio/rest/api/v2010/api_default.go#L48) signature wouldn't match the [docs](https://github.com/twilio/twilio-go/blob/template-update/twilio/rest/api/v2010/docs/DefaultApi.md#CreateAccount).

File: src/main/resources/twilio-go/api_doc.mustache [hunk 2]
```diff
@@ -14,11 +14,45 @@ Method | HTTP request | Description
...
 
 {{{summary}}}{{#notes}}
 
-{{{notes}}}{{/notes}}
+{{{unespacedNotes}}}{{/notes}}
+
+### Example
```
Comment (shwetha-manvinkurke) [inline src/main/resources/twilio-go/api_doc.mustache:23]: We have a [jira](https://issues.corp.twilio.com/browse/DI-1232) for this. I don't think the generated code examples our customizations.

File: src/main/resources/twilio-go/api_doc.mustache [hunk 3]
```diff
@@ -30,12 +64,12 @@ Name | Type | Description
...
+------------- | ------------- | -------------{{/-last}}{{/allParams}}{{#allParams}}
+{{^isPathParam}} **{{paramName}}** | {{#isContainer}}{{#isArray}}{{#items}}{{^isPrimitiveType}}{{^isFile}}[{{/isFile}}{{/isPrimitiveType}}**[]{{dataType}}**{{^isPrimitiveType}}{{^isFile}}]({{^baseType}}{{dataType}}{{/baseType}}{{dataType}}.md){{/isFile}}{{/isPrimitiveType}}{{/items}}{{/isArray}}{{#isMap}}{{#items}}{{^isPrimitiveType}}{{^isFile}}[{{/isFile}}{{/isPrimitiveType}}**map[string]{{dataType}}**{{^isPrimitiveType}}{{^isFile}}]({{^baseType}}{{dataType}}{{/baseType}}{{dataType}}.md){{/isFile}}{{/isPrimitiveType}}{{/items}}{{/isMap}}{{/isContainer}}{{^isContainer}}{{^isPrimitiveType}}{{^isFile}}[{{/isFile}}{{/isPrimitiveType}}**{{dataType}}**{{^isPrimitiveType}}{{^isFile}}]({{^baseType}}{{dataType}}{{/baseType}}{{dataType}}.md){{/isFile}}{{/isPrimitiveType}}{{/isContainer}} | {{description}}{{/isPathParam}}{{/allParams}}
```
Comment (shwetha-manvinkurke) [inline src/main/resources/twilio-go/api_doc.mustache:72]: This is going to [mess up the table formatting](https://github.com/twilio/twilio-go/blob/template-update/twilio/rest/api/v2010/docs/DefaultApi.md#other-parameters-1).

File: src/main/resources/twilio-go/model_doc.mustache [hunk 4]
```diff
@@ -1,15 +1,94 @@
...
+{{/vendorExtensions.x-is-one-of-interface}}
+{{^vendorExtensions.x-is-one-of-interface}}
+{{#vars}}**{{name}}** | {{^required}}Pointer to {{/required}}{{#isContainer}}{{#isArray}}{{#items}}{{^isPrimitiveType}}{{^isFile}}[{{/isFile}}{{/isPrimitiveType}}**[]{{dataType}}**{{^isPrimitiveType}}{{^isFile}}]({{dataType}}.md){{/isFile}}{{/isPrimitiveType}}{{/items}}{{/isArray}}{{#isMap}}{{#items}}{{^isPrimitiveType}}{{^isFile}}[{{/isFile}}{{/isPrimitiveType}}**map[string]{{dataType}}**{{^isPrimitiveType}}{{^isFile}}]({{^baseType}}{{dataType}}{{/baseType}}{{baseType}}.md){{/isFile}}{{/isPrimitiveType}}{{/items}}{{/isMap}}{{/isContainer}}{{^isContainer}}{{^isPrimitiveType}}{{^isFile}}{{^isDateTime}}[{{/isDateTime}}{{/isFile}}{{/isPrimitiveType}}**{{dataType}}**{{^isPrimitiveType}}{{^isFile}}{{^isDateTime}}]({{^baseType}}{{dataType}}{{/baseType}}{{baseType}}.md){{/isDateTime}}{{/isFile}}{{/isPrimitiveType}}{{/isContainer}} | {{description}} | {{^required}}[optional] {{/required}}{{#isReadOnly}}[readonly] {{/isReadOnly}}
+{{/vars}}
+{{/vendorExtensions.x-is-one-of-interface}}
+
+## Methods
```
Comment (shwetha-manvinkurke) [inline src/main/resources/twilio-go/model_doc.mustache:16]: These [docs](https://github.com/twilio/twilio-go/blob/template-update/twilio/rest/api/v2010/docs/ApiV2010AccountAddress.md#newapiv2010accountaddress) wouldn't match our generated code either

File: src/main/resources/twilio-go/api_doc.mustache [hunk 5]
```diff
@@ -30,7 +30,7 @@ Name | Type | Description
 
 ### Other Parameters
 
-Other parameters are passed through a pointer to a {{{nickname}}}Params struct
+Other parameters are passed through a pointer to a {{{nickname}}}Params struct via the builder pattern
```
[the remark this answers is not in the document]
Comment (thinkingserious) [reply src/main/resources/twilio-go/api_doc.mustache]: I don't think so, removed.

**Review by shwetha-manvinkurke — APPROVED** [review]

**Pull request merged**